AutoCollection/HeartBeat.ts

```diff
@@ -0,0 +1,138 @@
+import os = require("os");
+import TelemetryClient = require("../Library/TelemetryClient");
+import Constants = require("../Declarations/Constants");
+import Util = require("../Library/Util");
+import Config = require("../Library/Config");
+import Context = require("../Library/Context");
+
+class HeartBeat {
+
+    public static INSTANCE: HeartBeat;
+
+    private static AIMS_URI = "http://169.254.169.254/metadata/instance/compute";
+    private static AIMS_API_VERSION = "api-version=2017-12-01";
+    private static AIMS_FORMAT = "format=json";
+
+    private _collectionInterval: number = 900000;
+    private _client: TelemetryClient;
+    private _handle: NodeJS.Timer;
+    private _isEnabled: boolean;
+    private _isInitialized: boolean;
+    private _isVM: boolean = false;
+    private _vmData = <{[key: string]: string}>{};
+
+    constructor(client: TelemetryClient) {
+        if (!HeartBeat.INSTANCE) {
+            HeartBeat.INSTANCE = this;
+        }
+
+        this._isInitialized = false;
+        this._client = client;
+    }
+
+    public enable(isEnabled: boolean, config?: Config) {
```

Comment:
How is this being used? I do not believe heartbeat can be configurable.

Reply:
This is the pattern followed by all autocollectors in this SDK. If a particular autocollector should not be configurable, `setAutoCollectHeartbeat` should be omitted from `applicationinsights.ts` instead.